internal/operands/common-templates/reconcile_test.go

```diff
@@ -342,9 +344,28 @@ var _ = Describe("Common-Templates operand", func() {
 		It("should create common-template resources", func() {
 			_, err := operand.Reconcile(&request)
 			Expect(err).ToNot(HaveOccurred())
-			for _, template := range multiArchTemplates {
-				template.Namespace = namespace
-				ExpectResourceExists(&template, request)
+
+			for i := range multiArchTemplates {
+				foundTemplate := &templatev1.Template{}
+				Expect(request.Client.Get(request.Context, client.ObjectKey{
+					Name:      multiArchTemplates[i].Name,
+					Namespace: namespace,
+				}, foundTemplate)).To(Succeed())
+
+				Expect(foundTemplate.Labels).To(HaveKey(TemplateArchitectureLabel))
+				arch := foundTemplate.Labels[TemplateArchitectureLabel]
+
+				foundParam := false
+				for _, parameter := range foundTemplate.Parameters {
+					if parameter.Name == TemplateDataSourceParameterName {
+						Expect(parameter.Value).To(HaveSuffix("-" + arch))
+						foundParam = true
+						break
+					}
+				}
+				Expect(foundParam).To(BeTrue(), fmt.Sprintf("Template %s does not have parameter %s",
+					multiArchTemplates[i].Name,
+					TemplateDataSourceParameterName))
```

Comment:
nit: The `To` method performs the formatting for you, so you can do:
```suggestion
				Expect(foundParam).To(BeTrue(), "Template %s does not have parameter %s",
					multiArchTemplates[i].Name,
					TemplateDataSourceParameterName)
```

Alternatively, use the `BeTrueBecause` matcher; i.e.
```suggestion
				Expect(foundParam).To(BeTrueBecause("Template %s does not have parameter %s",
					multiArchTemplates[i].Name,
					TemplateDataSourceParameterName))
```

Reply:
Thanks, I will do it in fututre PR.